lib/active_record_extended/query_methods/with_cte.rb

```diff
@@ -135,17 +190,29 @@ def cte=(cte)
 
       # @return [Boolean]
       def with_values?
+        if WithCTE.cte_disabled?
+          return with_values_deprecated.present?
+        end
+
         !(cte.nil? || cte.empty?)
       end
 
       # @return [Array<Hash>]
-      def with_values
+      def with_values_deprecated
```

Comment:
Changed the name of this method, because it was causing circular references with `with_values` in active record query methods.